build/utils/scripts/dappnode_access_credentials.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+# PRIORITY: Wi-Wi > Avahi > VPN (Wireguard > OpenVpn)
+
+#############
+#0.VARIABLES#
+#############
+# Containers
+WIFI_CONTAINER="DAppNodeCore-wifi.dnp.dappnode.eth"
+WIREGUARD_CONTAINER="DAppNodeCore-wireguard.dnp.dappnode.eth"
+OPENVPN_CONTAINER="DAppNodeCore-vpn.dnp.dappnode.eth"
+HTTPS_CONTAINER="DAppNodeCore-https.dnp.dappnode.eth"
+# Credentials
+WIREGUARD_CREDS="docker exec $WIREGUARD_CONTAINER cat /config/peer_dappnode_admin/peer_dappnode_admin.conf"
+OPENVPN_CREDS="docker exec -i $OPENVPN_CONTAINER getAdminCredentials"
+WIFI_CREDS=$(cat /usr/src/dappnode/DNCORE/docker-compose-wifi.yml 2> /dev/null | grep 'SSID\|WPA_PASSPHRASE')
+# Endpoints
+DAPPNODE_ENDPOINT="http://my.dappnode"
+DAPPNODE_AVAHI_ENDPOINT="http://my.dappnode.local"
+
+#############
+#1.FUNCTIONS#
+#############
+
+# $1 Connection method $2 Credentials
+function create_connection_message () {
+  echo -e "\e[32mConnect to DAppNode through $1 using the following credentials:\e[0m\n$2\nVisit \e[4m$DAPPNODE_ENDPOINT\e"
+}
+
+function wifi_connection () {
+  # NOTE: network interface may be in use by host => $(cat /sys/class/net/${INTERFACE}/operstate)" !== "up")
+  # NOTE: wifi has a delay up to 1 min
+  INTERFACE=$(ls /sys/class/ieee80211/*/device/net 2> /dev/null)
+  [ "$(docker inspect -f '{{.State.Running}}' ${WIFI_CONTAINER} 2> /dev/null)" = "true" ] && [ ! -z $INTERFACE ] && create_connection_message "Wi-Fi" "$WIFI_CREDS" && exit 0 || echo "Wifi not available"
+}
+
+function avahi_connection () {
+  # Ping to avahi endpoint && avahi-daemon running => systemctl is-active avahi-daemon RETURNS "active" or "inactive"
+  ping -c 1 -w 10000 $DAPPNODE_AVAHI_ENDPOINT > /dev/null 2>&1 || { echo "Avahi-daemon not available" ; return ; }
```

Comment:
> -w deadline
Specify a timeout, in seconds, before ping exits regardless of how many packets have been sent or received. In this case ping does not stop after count packet are sent, it waits either for deadline expire or until count probes are answered or for some error notification from network.

a deadline of 10000 is pretty crazy, what the reasoning for `-c 1` and `-w 10000`? Could you detail why you chose specific parameters in a comment above the command?

Reply:
Not strong arguments.
- `-c` stands for count and it is the number of pings
- `-w` stands for timeout in milliseconds

1 count because should be more than enough to do only one ping
10 secs as timeout because we do not want to spend much time

Reply:
10 seconds not 10_000 seconds, the param is in seconds. Also 1 feels too low. Add comment in code too